Mathlib/Data/Fin/Tuple/Sort.lean

```diff
@@ -109,12 +111,52 @@ theorem monotone_sort (f : Fin n → α) : Monotone (f ∘ sort f) := by
 
 end Tuple
 
+theorem Fintype.card_fin_lt_nat (m g : ℕ) (h : g ≤ m) : Fintype.card {i : Fin m // i < g } = g := by
+  conv_rhs => rw [← Fintype.card_fin g]
+  apply Fintype.card_congr
+  exact ⟨ fun x => ⟨x, x.prop⟩, fun x => ⟨⟨x, (lt_of_lt_of_le (Fin.is_lt x) h)⟩, x.prop⟩,
+    fun x => by simp, fun x => by simp⟩
+
 namespace Tuple
 
 open List
 
 variable {n : ℕ} {α : Type*}
 
+/-- A sorted tuple with `m` elements and exactly `Fintype.card {i // f i ≤ a}` less than `a`, has
+the elements at the start, and vice versa -/
```

Comment:
I don't think I could understand this comment without the formal statement below. I admit it's hard to summarise the statement concisely but I suggest the following as an improvement:
```suggestion
/-- If `f₀ ≤ f₁ ≤ f₂ ≤ ⋯` is a sorted `m`-tuple of elements of `α`, then for any `j : Fin m` and
`a : α` we have `j < #{i | fᵢ ≤ a}` iff `fⱼ ≤ a`. -/
```